### PR TITLE
enlive

### LINKED ISSUE #37: replace selmer with enlive
Currently we use Selmer as a templating engine, but there are some limitations to what we can do with Selmer (for instance, it's very difficult to add custom functions to the templates and do data transformation from _within_ the templates (we need to do it before we pass our data structure into the template). 

Also, it is a bit cumbersome to use namespaced keywords within Selmer templates.

We should do a spike to see how it feels to use enlive instead of Selmer as a templating language. This might solve the problem that we currently have to drop the keywords for our view model, and would give us more flexibility to use clojure functions directly in the UI (e.g. being able to use the `bidi/route-for` function directly in the templates instead of having to construct URIs beforehand and then pass them into the data object).

### PR DESCRIPTION
**Pull request by joyheron**
Fixes #37

Moving templates over from Selmer to enlive one piece at a time.

This PR moves the existing templates over from Selmer to use enlive instead.
The main benefit here is that we can now use namespaced keywords and Clojure functions to transform our data within our templates (I personally think that it is a lot more pleasant to develop with).

### CODE REVIEW

File: src/spacy/app.clj [hunk 1]
```diff
@@ -28,18 +29,55 @@
...
+   ::please-wait "Please wait for others to present their session"
+   ::nobody-in-queue "There are currently no sessions in the queue"})
+
+(html/defsnippet up-next "templates/event/up-next.html"
+  [:up-next]
+  [{:keys [current-user is-next-up next-up
+           session-title statuses status]
+    :or   {session-title  (get-in next-up [:spacy.domain/session :spacy.domain/title])
+           statuses       (status-map session-title)
+           status         (current-status is-next-up next-up)}}]
```
Comment (joyheron) [inline src/spacy/app.clj:63]: @jstepien enlive seems to work pretty heavily with macros. I wasn't sure how to best create variables with a snippet scope, so I'm using `:or` here to do something that I would normally use a `let` for:

```clojure
(let [session-title (get-in next-up [:spacy.domain/session :spacy.domain/title])
      statuses (status-map session-title)
      status (current-status is-next-up next-up)]
  ;; do something with the variables here
  )
```

Do you know a better way to do this?

Comment (jstepien) [reply src/spacy/app.clj:63]: If I recall correctly, there's no easier way to introduce variables with a snippet scope.

You could consider introducing a function with that `let` you'd normally use. That function would build a map with all necessary data and pass it to the snippet. Something to the effect of:

```clojure
(defn up-next [{:keys [current-user is-next-up next-up ,,,]}
  (let [session-title (get-in next-up [:spacy.domain/session :spacy.domain/title])
        statuses (status-map session-title)
        status (current-status is-next-up next-up)
        values {,,,} ;; a map with all the snippet needs
        ]
    (up-next-snippet values))

(html/defsnippet up-next-snippet "templates/event/up-next.html"
  ,,,)
```

This leads to a simpler snippet. It gets all the data it needs and it's responsible only for correct templating.

Comment (joyheron) [reply src/spacy/app.clj]: Ah... thank you. That does seem more readable. I've modified this for the `up-next` snippet

**Comment by jstepien** [comment]
Looks great, thank you!

**Pull request merged**